src/Tweet.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+export default class Tweet extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      tweetText: '',
+      noOfCharsLeft: 50,
+      disabled: false
+    };
+
+    this.tweet = this.tweet.bind(this);
+    this.updateTweet = this.updateTweet.bind(this);
+  }
+
+  updateTweet(e) {
+    let noOfCharsLeft = 50 - e.target.value.length;
+
+    this.setState({
+      tweetText: e.target.value,
+      noOfCharsLeft: noOfCharsLeft,
+      disabled: noOfCharsLeft < 0
```

Comment:
When `noOfCharsLeft == 0`, the tweet can be posted, as the text has reached the maximum limit(50 in my case), but has not crossed it. 
`noOfCharsLeft = 50 - e.target.value.length`

For handling the case of no text, I'm changing this to :
      `disabled: noOfCharsLeft < 0 || !e.target.value.trim().length`